src/input.rs

```diff
@@ -437,6 +532,27 @@ mod tests {
         }
     }
 
+    test_clickstate! {
+        name: single_click,
+        initial_state: ClickState::None,
+        input: Event::MouseInput(ElementState::Pressed, MouseButton::Left),
+        end_state: ClickState::Click
+    }
+
+    test_clickstate! {
+        name: double_click,
+        initial_state: ClickState::Click,
+        input: Event::MouseInput(ElementState::Pressed, MouseButton::Left),
+        end_state: ClickState::DblClick
```

Comment:
I would expect the tests to check that the handler methods are called as appropriate as well.